File: server/src/models/status.ts
import Elysia, { t } from 'elysia';

export const Status = t.Object({
    id: t.Integer(),
    display: t.String(),
    description: t.String(),
    createdAt: t.Date(),
    color: t.String(),
    taskId: t.Integer(),
});

export const StatusDTO = t.Omit(Status, ['id', 'taskId']);

export const statusModels = new Elysia().model({
    'status.m': Status,
    'status.dto': StatusDTO,
});

File: server/internal/models/status.go
package models

import "time"

type Status struct {
	Id          int       `json:"id,omitempty" db:"id"`
	Display     string    `json:"display" validate:"required" db:"display"`
	Description string    `json:"description" db:"description"`
	CreatedAt   time.Time `json:"createdAt" db:"created_at"`
	Color       string    `json:"color" db:"color"`
}
